src/wombatLite.js

```diff
@@ -227,6 +228,67 @@ WombatLite.prototype.initAutoFetchWorker = function() {
   }
 };
 
+WombatLite.prototype.initHistoryOverrides = function() {
+  if (this.$wbwindow.self !== this.$wbwindow.top) {
+    return;
+  }
+
+  this.overrideHistoryFunc('pushState');
+  this.overrideHistoryFunc('replaceState');
+  var wombatLite = this;
+  this.$wbwindow.addEventListener('popstate', function(event) {
+    for (var i = 0; i < wombatLite.historyCB.length; i++) {
+      wombatLite.historyCB[i](wombatLite.$wbwindow.location.href,
+                              wombatLite.$wbwindow.document.title,
+                              "popstate",
+                              event.state);
+    }
+  });
+};
+
+
+WombatLite.prototype.overrideHistoryFunc = function(funcName) {
+  if (!this.$wbwindow.history) return undefined;
+  var orig_func = this.$wbwindow.history[funcName];
+  if (!orig_func) return undefined;
+
+  this.$wbwindow.history['_orig_' + funcName] = orig_func;
+  var wombat = this;
+
+  var rewrittenFunc = function histNewFunc(stateObj, title, url) {
+    orig_func.call(this, stateObj, title, url);
+
+    if (!url) {
+      return;
+    }
+
+    var origTitle = wombat.$wbwindow.document.title;
+
+    if (wombat.WBAutoFetchWorker) {
+      wombat.$wbwindow.setTimeout(function() {
+        if (!title && wombat.$wbwindow.document.title !== origTitle) {
+          title = wombat.$wbwindow.document.title;
+        }
+
+        wombat.WBAutoFetchWorker.fetchAsPage(wombat.$wbwindow.location.href, title);
+
+        for (var i = 0; i < wombat.historyCB.length; i++) {
```

Comment:
If only ever use one cb
```suggestion
        if (wombatLite.historyCB) {
      wombatLite.historyCB(
        wombatLite.$wbwindow.location.href,
        wombatLite.$wbwindow.document.title,
        'popstate',
        event.state
      );
    }
```